optuna/visualization/_edf.py

```
@@ -0,0 +1,138 @@
+
+            import math
+
+            import optuna
+
+
+            def ackley(x, y):
+                a = 20 * math.exp(-0.2 * math.sqrt(0.5 * (x ** 2 + y ** 2)))
+                b = math.exp(0.5 * (math.cos(2 * math.pi * x) + math.cos(2 * math.pi * y)))
+                return -a - b + math.e + 20
+
+
+            def objective(trial, low, high):
+                x = trial.suggest_float("x", low, high)
+                y = trial.suggest_float("y", low, high)
+                return ackley(x, y)
+
+
+            sampler = optuna.samplers.RandomSampler()
+
+            # Widest search space.
+            study0 = optuna.create_study(study_name="x=[0,5), y=[0,5)", sampler=sampler)
+            study0.optimize(lambda t: objective(t, 0, 5), n_trials=500)
+
+            # Narrower search space.
+            study1 = optuna.create_study(study_name="x=[0,4), y=[0,4)", sampler=sampler)
+            study1.optimize(lambda t: objective(t, 0, 4), n_trials=500)
+
+            # Narrowest search space but it doesn't include the global optimum point.
+            study2 = optuna.create_study(study_name="x=[1,3), y=[1,3)", sampler=sampler)
+            study2.optimize(lambda t: objective(t, 1, 3), n_trials=500)
+
+            optuna.visualization.plot_edf([study0, study1, study2])
+
+        .. raw:: html
+
+            <iframe src="../../_static/plot_edf.html"
+             width="100%" height="500px" frameborder="0">
+            </iframe>
+
+    Args:
+        study:
+            A target :class:`~optuna.study.Study` object.
+            You can pass multiple studies if you want to compare those EDFs.
+
+    Returns:
+        A :class:`plotly.graph_objs.Figure` object.
+    """
+
+    _imports.check()
+
+    if isinstance(study, list):
+        studies = study
+    else:
+        studies = [study]
+
+    return _get_edf_plot(studies)
+
+
+def _get_edf_plot(studies: List[Study]) -> "go.Figure":
+    if len(studies) == 0:
+        raise ValueError("No studies were given.")
```

Why don't you return the empty `go.Figure` in this case?

---

Honestly, there is no reason we should raise an error. I'll rewrite the code to return the `go.Figure` as you suggested. Thanks!

---

Fixed: https://github.com/optuna/optuna/pull/1482/commits/c5def17727d6153606741ffaf4461c767ea478bc